Fix unit test: using packaging.specifier instead of pkg_resources
See https://github.com/espressomd/espresso/issues/5127


diff -ruN espresso-4.2.2.orig/testsuite/python/unittest_decorators.py espresso-4.2.2/testsuite/python/unittest_decorators.py
--- espresso-4.2.2.orig/testsuite/python/unittest_decorators.py	2024-05-22 14:24:58.000000000 +0200
+++ espresso-4.2.2/testsuite/python/unittest_decorators.py	2025-07-25 12:18:36.595973547 +0200
@@ -20,6 +20,7 @@
 import sys
 import importlib
 import pkg_resources
+import packaging.specifiers
 import unittest
 
 import espressomd
@@ -72,7 +73,7 @@
         _module = importlib.import_module(module)
     except ImportError:
         return skipIfMissingModules(module)
-    if not pkg_resources.packaging.specifiers.SpecifierSet(
+    if not packaging.specifiers.SpecifierSet(
             version_requirement).contains(_module.__version__):
         return unittest.skip(
             "Skipping test: version requirement not met for module {}".format(module))
